wordpress-mobile/WordPress-Android
# [Reader Improvements] Post Details Header improvements

Part of #19085 

Implements the new design for the Post Details Header, using the new tag chips as well. The specs can be found here: zQnohyMpLzBzQ5jzMMKni3-fi-869%3A20299. It also hides the top bar "Visit Site" and "Share" items.

Note: this **only** touches the Post Details header implementation, it still does **not** change the Post body style to `sans-serif` **nor** update the bottom action bar. The follow/unfollow button was also not changes, as it's part of a different task.

Demo

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/ec9c15ed-71e4-4196-bf7a-eaab8565f656


To test:
1. Install and log in to Jetpack
2. Turn off the `ReaderImprovementsFeatureConfig` in Debug Settings
3. Go to Reader
4. Open some posts to see the post details page
5. **Verify** the old UI header still shows up as usual
6. Turn on the `ReaderImprovementsFeatureConfig` in Debug Settings
7. Open some posts to see the post details page
8. **Verify** the new UI for the header

Extra cases: try posts/sites without a site avatar (most from the Discover tab are like that due to a Discover endpoint limitation), and also try posts with Featured images.

## Regression Notes
1. Potential unintended areas of impact
Modifying the blog section/header in other places (such as posts in the post lists).

10. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

11. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
